shared/constants/types/fs.js

```diff
@@ -79,3 +95,52 @@ export const stringToPathType = (s: string): PathType => {
 export const pathTypeToString = (p: PathType): string => p
 export const pathConcat = (p: Path, s: string): Path =>
   p === '/' ? stringToPath('/' + s) : stringToPath(pathToString(p) + '/' + s)
+export const sortByToString = (s: SortBy): string => {
+  switch (s) {
+    case 'name':
+      return 'Name'
+    case 'time':
+      return 'Last Modified Time'
+    default:
+      throw new Error('invalid SortBy')
+  }
+}
+export const sortOrderToString = (s: SortOrder): string => {
+  switch (s) {
+    case 'asc':
+      return '⬆'
+    case 'desc':
+      return '⬇'
+    default:
+      throw new Error('invalid SortOrder')
+  }
+}
+
+type PathItemComparer = (a: PathItem, b: PathItem) => number
+
+const _sortByToAscComparer = (s: SortBy): PathItemComparer => (a: PathItem, b: PathItem): number => {
+  if (a.type === 'folder' && b.type !== 'folder') {
+    return -1
+  } else if (a.type !== 'folder' && b.type === 'folder') {
+    return 1
+  }
+
+  switch (s) {
+    case 'name':
+      return a.name < b.name ? -1 : 1
+    case 'time':
+      return a.lastModifiedTimestamp - b.lastModifiedTimestamp
+    case 'size':
+      return a.size - b.size
+    default:
+      throw new Error('invalid SortBy: ' + s)
+  }
+}
+
+const _sortByToDescComparer = (s: SortBy): PathItemComparer => {
+  const asc = _sortByToAscComparer(s)
+  return (a: PathItem, b: PathItem): number => -asc(a, b)
+}
+
+export const sortSettingToCompareFunction = (setting: SortSetting): Function =>
+  setting.sortOrder === 'desc' ? _sortByToDescComparer(setting.sortBy) : _sortByToAscComparer(setting.sortBy)
```

Comment:
i'd remove _sortByToDescComparer.  why have a function which does a check and then delegate to the other one and invert. just do it in this function
and just have this be like
```
export const sortSettingToCompareFunction = (setting: SortSetting): Function =>
  sortByToAscComparer(setting.sortBy) * (setting.sortOrder === 'desc' ? -1 : 1)
```

Reply:
Hmm `_sortByToAscComparer(setting.sortBy)` returns a function though. But fair point I'll merge them!